conan/api/subapi/cache.py

```diff
@@ -131,50 +130,48 @@ def save(self, package_list, tgz_path, no_source=False):
-                    pref_layout = cache.pkg_layout(pref)
-                    pkg_folder = pref_layout.package()
-                    folder = os.path.relpath(pkg_folder, cache_folder)
-                    folder = folder.replace("\\", "/")  # make win paths portable
-                    pref_bundle["package_folder"] = folder
-                    out.info(f"Saving {pref}: {folder}")
-                    tgz.add(os.path.join(cache_folder, folder), folder, recursive=True)
-                    if os.path.exists(pref_layout.metadata()):
-                        metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
-                        metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
-                        pref_bundle["metadata_folder"] = metadata_folder
-                        out.info(f"Saving {pref} metadata: {metadata_folder}")
-                        tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
-                                recursive=True)
-            serialized = json.dumps(package_list.serialize(), indent=2)
-            info = tarfile.TarInfo(name="pkglist.json")
-            data = serialized.encode('utf-8')
-            info.size = len(data)
-            tgz.addfile(tarinfo=info, fileobj=BytesIO(data))
-            tgz.close()
+                    tar_files.update({f"{recipe_folder}/{f}": path})
+            path = os.path.join(cache_folder, recipe_folder, DOWNLOAD_EXPORT_FOLDER, METADATA)
+            if os.path.exists(path):
+                tar_files.update({f"{recipe_folder}/{DOWNLOAD_EXPORT_FOLDER}/{METADATA}": path})
+
+            for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():
+                pref_layout = cache.pkg_layout(pref)
+                pkg_folder = pref_layout.package()
+                folder = os.path.relpath(pkg_folder, cache_folder)
+                folder = folder.replace("\\", "/")  # make win paths portable
+                pref_bundle["package_folder"] = folder
+                out.info(f"Saving {pref}: {folder}")
+                tar_files.update({folder: os.path.join(cache_folder, folder)})
+
+                if os.path.exists(pref_layout.metadata()):
+                    metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
+                    metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
+                    pref_bundle["metadata_folder"] = metadata_folder
+                    out.info(f"Saving {pref} metadata: {metadata_folder}")
+                    tar_files.update({metadata_folder: os.path.join(cache_folder, metadata_folder)})
+
+        serialized = json.dumps(package_list.serialize(), indent=2)
+        pkglist_path = os.path.join(cache_folder, "pkglist.json")
+        with open(pkglist_path, "w") as file_handler:
+            file_handler.write(serialized)
+        tar_files.update({"pkglist.json": pkglist_path})
```

Comment:
this was avoided on purpose, to avoid polluting the Conan cache with an extra ``pkglist.json``, that would be left there dirty. Maybe it can be put in another temporary folder? How to avoid the unzipping of such a file on the target ``conan cache restore``?

Reply:
Okey, I thought about it. Already fixed by creating the pkglist in a temporary